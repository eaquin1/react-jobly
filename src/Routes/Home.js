import React, {useContext} from "react"
import img from "../Static/jobly.png"
import {Link} from "react-router-dom"
import UserContext from "../UserContext"
function Home() {
    const {currentUser} = useContext(UserContext)
    return (
        <div>
        <h1>Jobly</h1>
        <img src={img} alt="People working" style={{width: "900px"}}></img>
        {currentUser ? <h2>Welcome Back</h2> : <Link to="Login" style={{display: "block", fontSize: "3em"}}>Login</Link>}
        
        </div>
    )
    }

export default Home