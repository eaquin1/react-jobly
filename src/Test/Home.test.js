import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Home from '../Routes/Home';
import { UserProvider } from '../testUtils';

describe('Testing the Home component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider>
          <Home />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <Home />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when logged out', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider currentUser={null}>
          <Home />
        </UserProvider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});