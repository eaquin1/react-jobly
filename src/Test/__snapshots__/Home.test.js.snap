// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Home component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Jobly
    </h1>
    <img
      alt="People working"
      src="jobly.png"
      style="width: 900px;"
    />
    <h2>
      Welcome Back
    </h2>
  </div>
</DocumentFragment>
`;

exports[`Testing the Home component matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Jobly
    </h1>
    <img
      alt="People working"
      src="jobly.png"
      style="width: 900px;"
    />
    <a
      class="btn btn-lg btn-secondary"
      href="/Login"
    >
      Login
    </a>
  </div>
</DocumentFragment>
`;
